telethon/network/mtproto_sender.py

```diff
@@ -158,8 +158,11 @@ def _decode_msg(self, body):
             msg_key = reader.read(16)
 
             key, iv = utils.calc_key(self.session.auth_key.key, msg_key, False)
+            data = reader.read(len(body) - reader.tell_position())
+            if len(data) % 16 != 0:
```

Comment:
I'm still not sure about this… How can it happen? I would rather we found the root of the problem.